homeassistant/components/bmw_connected_drive/sensor.py

```diff
@@ -1,51 +1,101 @@
+    "remaining_range_fuel": ["mdi:map-marker-distance", LENGTH_KILOMETERS, True],
+    "max_range_electric": ["mdi:map-marker-distance", LENGTH_KILOMETERS, True],
+    "remaining_fuel": ["mdi:gas-station", VOLUME_LITERS, True],
+    # LastTrip attributes
+    "average_combined_consumption": [
+        "mdi:flash",
+        f"{ENERGY_KILO_WATT_HOUR}/100{LENGTH_KILOMETERS}",
+        True,
+    ],
+    "average_electric_consumption": [
+        "mdi:power-plug-outline",
+        f"{ENERGY_KILO_WATT_HOUR}/100{LENGTH_KILOMETERS}",
+        True,
+    ],
+    "average_recuperation": [
+        "mdi:recycle-variant",
+        f"{ENERGY_KILO_WATT_HOUR}/100{LENGTH_KILOMETERS}",
+        True,
+    ],
+    "electric_distance": ["mdi:map-marker-distance", LENGTH_KILOMETERS, True],
+    "saved_fuel": ["mdi:fuel", VOLUME_LITERS, False],
+    "total_distance": ["mdi:map-marker-distance", LENGTH_KILOMETERS, True],
 }
 
 ATTR_TO_HA_IMPERIAL = {
-    "mileage": ["mdi:speedometer", LENGTH_MILES],
-    "remaining_range_total": ["mdi:map-marker-distance", LENGTH_MILES],
-    "remaining_range_electric": ["mdi:map-marker-distance", LENGTH_MILES],
-    "remaining_range_fuel": ["mdi:map-marker-distance", LENGTH_MILES],
-    "max_range_electric": ["mdi:map-marker-distance", LENGTH_MILES],
-    "remaining_fuel": ["mdi:gas-station", VOLUME_GALLONS],
-    "charging_time_remaining": ["mdi:update", TIME_HOURS],
-    "charging_status": ["mdi:battery-charging", None],
+    "mileage": ["mdi:speedometer", LENGTH_MILES, True],
+    "remaining_range_total": ["mdi:map-marker-distance", LENGTH_MILES, True],
+    "remaining_range_electric": ["mdi:map-marker-distance", LENGTH_MILES, True],
+    "remaining_range_fuel": ["mdi:map-marker-distance", LENGTH_MILES, True],
+    "max_range_electric": ["mdi:map-marker-distance", LENGTH_MILES, True],
+    "remaining_fuel": ["mdi:gas-station", VOLUME_GALLONS, True],
+    # LastTrip attributes
+    "average_combined_consumption": [
+        "mdi:flash",
+        f"{ENERGY_KILO_WATT_HOUR}/100{LENGTH_MILES}",
+        True,
+    ],
+    "average_electric_consumption": [
+        "mdi:power-plug-outline",
+        f"{ENERGY_KILO_WATT_HOUR}/100{LENGTH_MILES}",
+        True,
+    ],
+    "average_recuperation": [
+        "mdi:recycle-variant",
+        f"{ENERGY_KILO_WATT_HOUR}/100{LENGTH_MILES}",
+        True,
+    ],
+    "electric_distance": ["mdi:map-marker-distance", LENGTH_MILES, True],
+    "saved_fuel": ["mdi:fuel", VOLUME_GALLONS, False],
+    "total_distance": ["mdi:map-marker-distance", LENGTH_MILES, True],
+}
+
+ATTR_TO_HA_GENERIC = {
+    "charging_time_remaining": ["mdi:update", TIME_HOURS, True],
+    "charging_status": ["mdi:battery-charging", None, True],
     # No icon as this is dealt with directly as a special case in icon()
-    "charging_level_hv": [None, PERCENTAGE],
+    "charging_level_hv": [None, PERCENTAGE, True],
+    # LastTrip attributes
+    "date_utc": ["mdi:calendar", None, True],
```

Comment:
I think we can use device class timestamp for this sensor. Then we should return the datetime as an isoformatted string.

https://docs.python.org/3/library/datetime.html#datetime.datetime.isoformat

https://developers.home-assistant.io/docs/core/entity/sensor#available-device-classes

Please implement the `device_class` property in the entity.